embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/storage/EmbraceStorageService.kt

```diff
@@ -0,0 +1,55 @@
+package io.embrace.android.embracesdk.storage
+
+import io.embrace.android.embracesdk.injection.CoreModule
+import java.io.File
+import java.io.FilenameFilter
+
+internal class EmbraceStorageService(
+    private val coreModule: CoreModule
+) : StorageService {
+
+    override val cacheDirectory: Lazy<File> = lazy {
```

Comment:
```suggestion
    override val cacheDirectory: File by lazy {
```

Using [property delegation](https://kotlinlang.org/docs/delegated-properties.html) for both of these properties would avoid having to use `cacheDirectory.value` everywhere

Reply:
Done.